substrafl/remote/register/register.py

```diff
@@ -73,6 +73,10 @@
 # Install requirements
 RUN python{python_version} -m pip install --no-cache-dir -r requirements.txt
 
+USER root
+RUN apt-get purge -y --auto-remove build-essential *-dev
+USER user
+
```

Comment:
Are we forced to do it in last step ? Do we need these libs to setup the image ?

Reply:
I think we may need these when we compile python module

Reply:
Okay. And using the root user might raise some alert no ?

Reply:
What do you mean by alert? What kind? To who?

Reply:
On the builder while building the image ? I'm not sure I just want to check that using a root user while building will not trigger anything 😇

Reply:
This part (before switching to the new user) is already run as root
```
_CPU_BASE_IMAGE = """
FROM python:{python_version}-slim

# update image
RUN apt-get update -y
"""

_GPU_BASE_IMAGE = """
FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04

# update image & install Python
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update -y\
    && apt-get install -y software-properties-common\
    && add-apt-repository -y ppa:deadsnakes/ppa\
    && apt-get -y upgrade\
    && apt-get install -y python{python_version} python{python_version}-venv python3-pip\
    && apt-get clean\
    && rm -rf /var/lib/apt/lists/*

"""
```
what kind of test would you like to run to check if no warning is raised?

Reply:
Okay you're right :) no that's ok then thanks ! we can just launch an e2e test, like mnist, on it :)